File: Rotate for a Max/Kotlin/solution_1.kt
package maxrot

fun maxRot(n: Long): Long {
    var digits = n.toString()
    var maxVal = n
    for (i in digits.indices) {
        val c = digits[i]
        digits = digits.removeRange(i, i + 1) + c
        val rotated = digits.toLong()
        if (rotated > maxVal) {
            maxVal = rotated
        }
    }
    return maxVal
}

File: Rotate for a Max/C++/solution_1.cpp
#include <iostream>
#include <string>

class MaxRotate
{
public:
  static long long maxRot(long long n)
  {
    std::string digits = std::to_string(n);
    long long max_val = n;
    for (int i = 0; i < digits.length(); i++)
    {
        char c = digits[i];
        digits.erase(digits.begin() + i);
        digits.push_back(c);
        long long rotated = std::stoll(digits);
        if (rotated > max_val)
        {
            max_val = rotated;
        }
    }
    return max_val;
  }
};
